util/ranger/testdata/ranger_suite_out.json

```diff
@@ -387,5 +387,146 @@
         ]
       }
     ]
+  },
+  {
+    "Name": "TestIndexRangeForBit",
+    "Cases": [
+      {
+        "SQL": "select * from t;",
+        "Plan": [
+          "TableReader_5 4.00 root partition:all data:TableFullScan_4",
+          "└─TableFullScan_4 4.00 cop[tikv] table:t keep order:false"
+        ],
+        "Result": [
+          "\u0000 0",
+          "\u0001 -1",
+          "\u0001 1",
+          "\u0001 3"
+        ]
+      },
+      {
+        "SQL": "select * from t where a = 0;",
+        "Plan": [
+          "Selection_5 3.20 root  eq(test.t.a, 0)",
+          "└─TableReader_7 4.00 root partition:p0 data:TableFullScan_6",
+          "  └─TableFullScan_6 4.00 cop[tikv] table:t keep order:false"
+        ],
+        "Result": [
+          "\u0000 0"
+        ]
+      },
+      {
+        "SQL": "select * from t where a = 0 or a = 4;",
+        "Plan": [
+          "Selection_5 3.20 root  or(eq(test.t.a, 0), eq(test.t.a, 4))",
+          "└─TableReader_7 4.00 root partition:p0 data:TableFullScan_6",
+          "  └─TableFullScan_6 4.00 cop[tikv] table:t keep order:false"
+        ],
+        "Result": [
+          "\u0000 0"
+        ]
+      },
+      {
+        "SQL": "select * from t where a = 1;",
+        "Plan": [
+          "Selection_5 3.20 root  eq(test.t.a, 1)",
+          "└─TableReader_7 4.00 root partition:p1 data:TableFullScan_6",
+          "  └─TableFullScan_6 4.00 cop[tikv] table:t keep order:false"
+        ],
+        "Result": [
+          "\u0001 -1",
+          "\u0001 1",
+          "\u0001 3"
+        ]
+      },
+      {
+        "SQL": "select * from t where a = -1;",
+        "Plan": [
+          "Selection_5 3.20 root  eq(test.t.a, -1)",
+          "└─TableReader_7 4.00 root partition:dual data:TableFullScan_6",
+          "  └─TableFullScan_6 4.00 cop[tikv] table:t keep order:false"
```

Comment:
It's weird for `partition:dual`. Can we change it to `TableDual`?

Reply:
I think we can do this in next PR. What's your opinion? @lzmhhh123

Reply:
Okay.